### PR TITLE
Updating Download/Delete/Play operations per Board review.

### PR DESCRIPTION
**Pull request by cochi2**
This updates:
* Download/Delete requests' parameters' names.
* PlayMedia -> Play method name.
* Adds specific `FileSource` parameter to Play method.

### CODE REVIEW

**Review by minwoolee-msft — APPROVED** [review]

**Comment by minwoolee-msft** [comment]
It seems like some of analyze is failing. This must be passed to have this merged. Try this:
1. Go to the SDK folder (...\azure-sdk-for-net\) with powershell 7.x
2. Run this command: **eng\scripts\CodeChecks.ps1 -ServiceDirectory communication/Azure.Communication.CallingServer**
3. If it fails, try run this command: **eng\scripts\Export-API.ps1 -ServiceDirectory communication/Azure.Communication.CallingServer**
4. You might get some error, try re-generate with autorest and try step 2 and 3 again. 
5. Once Step #2 returns OK, it should also pass the github checks

**Comment by cochi2** [comment]
> It seems like some of analyze is failing. This must be passed to have this merged. Try this:
> 
> 1. Go to the SDK folder (...\azure-sdk-for-net) with powershell 7.x
> 2. Run this command: **eng\scripts\CodeChecks.ps1 -ServiceDirectory communication/Azure.Communication.CallingServer**
> 3. If it fails, try run this command: **eng\scripts\Export-API.ps1 -ServiceDirectory communication/Azure.Communication.CallingServer**
> 4. You might get some error, try re-generate with autorest and try step 2 and 3 again.
> 5. Once Step 2 returns OK, it should also pass the github checks

Thank you, @minwoolee-msft, it worked :)

File: sdk/communication/Azure.Communication.CallingServer/api/Azure.Communication.CallingServer.netstandard2.0.cs [hunk 1]
```diff
@@ -34,10 +34,10 @@ public partial class CallConnection
...
-        public virtual System.Threading.Tasks.Task<Azure.Response> PlayMediaAsync(Azure.Communication.CallingServer.PlaySource playSource, System.Collections.Generic.IEnumerable<Azure.Communication.CommunicationIdentifier> playTo, System.Threading.CancellationToken cancellationToken = default(System.Threading.CancellationToken)) { throw null; }
-        public virtual Azure.Response PlayMediaToAll(Azure.Communication.CallingServer.PlaySource playSource, System.Threading.CancellationToken cancellationToken = default(System.Threading.CancellationToken)) { throw null; }
-        public virtual System.Threading.Tasks.Task<Azure.Response> PlayMediaToAllAsync(Azure.Communication.CallingServer.PlaySource playSource, System.Threading.CancellationToken cancellationToken = default(System.Threading.CancellationToken)) { throw null; }
+        public virtual Azure.Response Play(Azure.Communication.CallingServer.FileSource fileSource, System.Collections.Generic.IEnumerable<Azure.Communication.CommunicationIdentifier> playTo, System.Threading.CancellationToken cancellationToken = default(System.Threading.CancellationToken)) { throw null; }
```
Comment (minwoolee-msft) [inline sdk/communication/Azure.Communication.CallingServer/api/Azure.Communication.CallingServer.netstandard2.0.cs:37]: For this section, I thought you wanted to accept PlaySource? for accepting different types of sources (FileSource, TTS-Source, etc)

Comment (cochi2) [reply sdk/communication/Azure.Communication.CallingServer/api/Azure.Communication.CallingServer.netstandard2.0.cs:37]: Hey @minwoolee-msft, happy to discuss this: We received the comment in the reviews that PlaySource naming is not great, and some others were confused with what we are actually playing.
I thought that having these methods help to the reading, in the semantics of "Play this FileSource", "Play this TextSource", etc. So we can cut on the naming problems. 
We are doing something similar on download with `downloadTo(String filePath...)` and `downloadTo(Stream stream, ...` and I think that it helps.

Please share your thoughts on this :)

Comment (minwoolee-msft) [reply sdk/communication/Azure.Communication.CallingServer/api/Azure.Communication.CallingServer.netstandard2.0.cs:37]: From API, I see both Playsource and filesource - the playsource has PlaySourceType (currenlty only File is there), and it has FileSource as one of its properties. So my assumption here was that PlaySource is the wrapper model (abstract) and the contoso would pass in FileSource, or other source to it. 
I am afraid if we do pass just the filesource here, what is the purpose of having playsource in the API? can we just drop playsource?

Comment (cochi2) [reply sdk/communication/Azure.Communication.CallingServer/api/Azure.Communication.CallingServer.netstandard2.0.cs:37]: Well, `PlaySource` will become useful when we add more sources to the mix. 
Maybe I can move it to be an internal object, so it works to refactor code in the future.

**Pull request merged**